docs/gcp-worker-guide.md

```diff
@@ -0,0 +1,337 @@
+- [Docker](https://www.docker.com/get-started/) installed on your local machine.
+- A Prefect server instance. You can sign up for a forever free [Prefect Cloud Account](https://app.prefect.cloud/) or, alternatively, self-host a [Prefect server](https://docs.prefect.io/latest/guides/host/).
+
+### Step 1. Create a Google Cloud service account
+First, open a terminal or command prompt on your local machine where `gcloud` is installed. If you haven't already authenticated with `gcloud`, run the following command and follow the instructions to log in to your GCP account.
+
+```bash
+gcloud auth login
+```
+
+Next, you'll set your project where you'd like to create the service account. Use the following command and replace `<PROJECT_ID>` with your GCP project's ID.
+
+```bash
+gcloud config set project <PROJECT-ID>
+```
+
+For example, if your project's ID is `prefect-project` the command will look like this:
+
+```bash
+gcloud config set project prefect-project
+```
+
+Now you're ready to make the service account. To do so, you'll need to run this command:
+
+```bash
+gcloud iam service-accounts create <SERVICE-ACCOUNT-NAME> --display-name="<DISPLAY-NAME>"
+```
+
+Here's an example of the command above which you can use which already has the service account name and display name provided. An additional option to describe the service account has also been added:
+
+```bash
+gcloud iam service-accounts create prefect-service-account \
+    --description="service account to use for the prefect worker" \
+    --display-name="prefect-service-account"
+```
+
+The last step of this process is to make sure the service account has the proper permissions to execute flow runs as Cloud Run jobs.
+Run the following commands to grant the necessary permissions:
+
+```bash
+gcloud projects add-iam-policy-binding <PROJECT-ID> \
+    --member="serviceAccount:<SERVICE-ACCOUNT-NAME>@<PROJECT-ID>.iam.gserviceaccount.com" \
+    --role="roles/iam.serviceAccountUser"
+```
+```bash
+gcloud projects add-iam-policy-binding <PROJECT-ID> \
+    --member="serviceAccount:<SERVICE-ACCOUNT-NAME>@<PROJECT-ID>.iam.gserviceaccount.com" \
+    --role="roles/run.admin"
+```
+
+### Step 2. Create a Cloud Run work pool
+Let's walk through the process of creating a Cloud Run work pool.
+
+#### Create a GCP Credentials Block
+You'll need to create a GCP Credenitals block to manage authentication wth GCP. This block will be referenced in the base job template of your work pool.
+
+The block created in this guide will contain the JSON key for the service account created in the previous step.
+To get the JSON key, paste the following command into your terminal.
+```bash
+gcloud iam service-accounts keys create my_key.json \
+    --serviceAccount:<SERVICE_ACCOUNT_NAME>@<PROJECT_ID>.iam.gserviceaccount.com
+```
```

Comment:
@biancaines Hi, I am trying out the guide at the moment to setup prefect with cloud run for my org. Thank you so much, it is such a valuable resource! :blush:

This command however did not work for me, because `--serviceAccount` is not recognized as a command:

```
ERROR: (gcloud.iam.service-accounts.keys.create) unrecognized arguments: --serviceAccount:<my-account-name>@<my-project>.iam.gserviceaccount.com
```

With the help of the GCP Documentation [here](https://cloud.google.com/iam/docs/keys-create-delete), the following command worked for me:
```bash
 gcloud iam service-accounts keys create my_key.json \
    --iam-account=<SERVICE_ACCOUNT_NAME>@<PROJECT_ID>.iam.gserviceaccount.com
```

Reply:
Hello @bjarneschroeder ! Thank you for you kind words, and for providing the command you used! I'll give this a try as well, and either swap out the command or create a note that it can be used as an alternative.